supervisor/erratum_handler.py

```diff
@@ -56,26 +64,41 @@ def erratum_needs_attention(erratum_id: int) -> bool:
     return issue is not None
 
 
-def erratum_all_issues_are_release_pending(
-    erratum: Erratum, issue_cache: dict[str, Issue]
-) -> bool:
+def all_issues_are_release_pending(issues: dict[str, Issue] | list[Issue]) -> bool:
+    return all(
+        issue.status == IssueStatus.RELEASE_PENDING
+        for issue in (issues.values() if isinstance(issues, dict) else issues)
+    )
+
+
+def erratum_get_issues(
+    erratum: Erratum, *, issue_cache: dict[str, Issue] = {}, full: bool = False
+):
     # The errata data we fetch from errata-tool includes details
     # of the errata beyond the ID - in particular it has the status
     # of the issue - but due to a bug in errata tool that is returning
     # stale data, so we need to fetch the status from JIRA directly.
     # https://issues.redhat.com/browse/RHELWF-13481
 
-    # Start with the statuses we have in the cache
-    statuses = {
-        key: issue.status if (issue := issue_cache.get(key)) else None
-        for key in erratum.jira_issues
-    }
-    # Then fetch any that were missing
-    to_fetch = [key for key, status in statuses.items() if status is None]
-    if to_fetch:
-        statuses.update(get_issues_statuses(to_fetch))
+    issue_keys = erratum.jira_issues
+    missing_issues: dict[str, Issue] = dict()
+    for issue_key in issue_keys:
+        if issue_key not in issue_cache:
+            missing_issues[issue_key] = get_issue(issue_key, full=full)
+
+    return issue_cache | missing_issues
```

Comment:
This is an unexpected return! I think you just want to return a list of issues looked up from the cache if available, otherwise fetched using get_issue().